packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -325,25 +356,94 @@ function mouseUpdate(event: MouseEvent) {
 	prevY = event.y;
 }
 
-onMounted(() => {
+function amrToTemplates() {
+	kernelManager.sendMessage('amr_to_templates', {}).on('amr_to_templates_response', (d) => {
+		// FIXME: Model templates are passed no junctions yet
+
+		// Insert template card data into template models
+		let yPos = 100;
+		const templateModelsWithCards = d.content.templates.map((modelTemplate: any) => {
+			modelTemplate.metadata.templateCard = {
+				id: modelTemplate.header.name,
+				name: modelTemplate.header.name,
+				x: 100,
+				y: yPos
+			} as ModelTemplateCard;
+
+			yPos += 200;
+			return modelTemplate;
+		});
+
+		templateModelsWithCards.forEach((templateModel: any) =>
+			modelTemplatingService.addCard(modelTemplateEditor.value, templateModel)
+		);
+	});
+}
+
+watch(
+	() => props.model,
+	async () => {
+		if (props.model) {
+			// Create flattened view of model
+			const flattenedModel: any = cloneDeep(props.model);
+			flattenedModel.metadata.templateCard = {
+				id: props.model.id,
+				name: props.model.header.name,
+				x: 100,
+				y: 100
+			};
+			console.log(flattenedModel);
+			modelTemplatingService.addCard(flatModelTemplateEditor.value, flattenedModel);
+
+			// Initialize beaker kernel
+			try {
+				const context = {
+					context: 'mira_model',
+					language: 'python3',
+					context_info: {
+						id: props.model.id
+					}
+				};
+				await kernelManager.init('beaker_kernel', 'Beaker Kernel', context);
```

Comment:
If you anticipate the possibility of re-initializing the kernelManager, you should calling a "shutdown" before